src/main/java/net/glowstone/block/blocktype/BlockTnt.java

```diff
@@ -7,28 +7,40 @@
 import org.bukkit.Material;
 import org.bukkit.Sound;
 import org.bukkit.World;
-import org.bukkit.block.Block;
 import org.bukkit.block.BlockFace;
 import org.bukkit.entity.EntityType;
 import org.bukkit.inventory.ItemStack;
 
 public class BlockTnt extends BlockType {
 
     /**
-     * Convert a TNT block into a primed TNT entity.
+     * Convert a TNT block into a primed TNT entity with the player who ignited the TNT.
      *
      * @param tntBlock The block to ignite.
      * @param ignitedByExplosion True if another explosion caused this ignition.
+     * @param player The player who ignited the TNT.
      */
-    public static void igniteBlock(Block tntBlock, boolean ignitedByExplosion) {
+    public static void igniteBlock(
+        GlowBlock tntBlock, boolean ignitedByExplosion, GlowPlayer player) {
```

Comment:
Why do you have to narrow tntBlock from Block to GlowBlock? Things like that make unit testing more complicated.

Reply:
It's more consistent within the class. All other methods use Glowblock.

Reply:
I agree with Pr0methean, it should be kept like `Block`, unless there is a specific reason for doing it otherwise. The other usages of `GlowBlock` as a method parameter are not reasons for this one to use it.